src/Compilers/CSharp/Portable/Symbols/ParameterSymbol.cs

```diff
@@ -385,6 +385,56 @@ internal sealed override ObsoleteAttributeData ObsoleteAttributeData
 
         internal abstract bool IsCallerMemberName { get; }
 
+        /// <summary>
+        /// If the method returns false, then we can infer that the argument used for this parameter was not null.
+        /// </summary>
+        internal abstract bool NotNullWhenFalse { get; }
+
+        /// <summary>
+        /// When the method returns, then we can infer that the argument used for this parameter was not null.
+        /// </summary>
+        internal abstract bool EnsuresNotNull { get; }
```

Comment:
Should these two properties be combined, with a single `enum` value?
```
[Flags] enum EnsuresNullability { None = 0, NotNullWhenFalse = 1, NotNullWhenTrue = 2 }
```

Reply:
Should the two attributes be combined into a single `[Ensures(...)]` attribute?

Reply:
From our discussion, I'll keep separate attributes for the moment. We'll discuss with LDM.

---
In reply to: [187089388](https://github.com/dotnet/roslyn/pull/26656#discussion_r187089388) [](ancestors = 187089388)